### PR TITLE
libgpod: drop pygobject2

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Let's drop libgpod using pygobject2.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
Weird, they still have the pygobject2 import in the code:

https://github.com/gpodder/gpodder/blob/30586eb29cb42b6aad33deeb9ef8370deff37af9/src/gpodder/sync.py#L47

**Review by jtojnar — APPROVED** [review]

File: pkgs/development/libraries/libgpod/default.nix [hunk 1]
```diff
@@ -1,18 +1,20 @@
-{stdenv, lib, fetchurl, gettext, perlPackages, intltool, pkgconfig, glib,
+{ stdenv, lib, fetchurl, gettext, perlPackages, intltool, pkgconfig, glib,
   libxml2, sqlite, zlib, sg3_utils, gdk-pixbuf, taglib,
-  libimobiledevice, python3Packages, mutagen,
+  libimobiledevice, mutagen,
   monoSupport ? false, mono, gtk-sharp-2_0
 }:
 
-let
-  inherit (python3Packages) python pygobject2;
```
Comment (jtojnar) [inline pkgs/development/libraries/libgpod/default.nix]: Though this is actually pygobject2 on Python 3.

Comment (worldofpeace) [reply pkgs/development/libraries/libgpod/default.nix]: Oh, I didn't even notice this. And pygobject2 did support python3.

File: pkgs/development/libraries/libgpod/default.nix [hunk 2]
```diff
@@ -23,7 +25,7 @@ in stdenv.mkDerivation rec {
   dontStrip = true;
 
   propagatedBuildInputs = [ glib libxml2 sqlite zlib sg3_utils
-    gdk-pixbuf taglib libimobiledevice python pygobject2 mutagen ];
+    gdk-pixbuf taglib libimobiledevice  mutagen ];
```
Comment (jonringer) [inline pkgs/development/libraries/libgpod/default.nix]: I feel like a lot of these shouldn't be propagated at all. Along with `dontStrip = true;` it inflates the closure size by a lot:

```
$ nix path-info -Sh ./result
/nix/store/1g3ipk8jqc7brx7bivhi52fqylw9v1fd-libgpod-0.8.3        417.6M
$ nix path-info -Sh ./result
/nix/store/5ha9yd4g7wik1qq0bawb57cb772k4ynp-libgpod-0.8.3         78.0M
```

Comment (worldofpeace) [reply pkgs/development/libraries/libgpod/default.nix]: A lot of which are in `Requires`, so they have to be propagated for pkg-config to work correct.
But indeed it's overlinking, and there's a patch for it https://src.fedoraproject.org/rpms/libgpod/blob/master/f/libgpod-0.8.2-pkgconfig_overlinking.patch. But given I don't maintain this package, it doesn't get this kind of improvement in the scope of the PR :smile:

Comment (jonringer) [reply pkgs/development/libraries/libgpod/default.nix]: only 4 of them are
```
Requires: glib-2.0 >= 2.8.0 gobject-2.0 gdk-pixbuf-2.0 libimobiledevice-1.0
```

Comment (jonringer) [reply pkgs/development/libraries/libgpod/default.nix]: but I understand not want to shine an expression to a mirror-like shine when you're not very invested in it. I'm with with the current changes.

**Pull request merged**